src/gt4py/backend/pyext_builder.py

```diff
@@ -51,6 +51,7 @@ def get_gt_pyext_build_opts(
         ],
         nvcc=[
             "-std=c++14",
+            "-arch=sm_35",
```

Comment:
Is this required?

Reply:
Yes, this is required for the `__ldg` intrinsic that is used by the `dawn:cuda` backend.

Reply:
Can we make this read the `CUDA_ARCH` env variable so it can be overridden with a newer architecture flag as possible?

Reply:
Implemented.

Reply:
If not explicitly set in the environment, we should get it at runtime, e.g.: 
```
def get_cuda_compute_capability():
    try:
        import cupy as cp

        return cp.cuda.Device(0).compute_capability
    except:
        return None
```

Reply:
Also, it would be nice if an error is raised if the device is too old for that backend.

Reply:
I agree with Linus about detecting the compute capability at runtime if not explicitly provided. Actually, it might be better to raise an exception if the detection does not work to avoid compilations without setting a specific architecture. Something like:
```python
# config.py

"cuda_arch": os.environ.get("CUDA_ARCH", None)
```
and then in the `pyext_builder`:
```python
cuda_arch = gt_config.build_settings["cuda_arch"] or get_cuda_compute_capability()
if not cuda_arch:
    raise Exception(...) # Cuda architecture not set....
...

             "-arch={}".format(cuda_arch), 
```